### ISSUE TITLE
unique identifiers

### ISSUE THREAD

**Issue by Park-Patrick** [afids/afids-validator#48]
Some thoughts brought up today were:
- email
- full login capabilities
- security 
- time to implement

My tentative suggestion is to have a userid (i.e. jsmith5) field that the uploader can fill in before adding data. This would mean that anyone can add to anyone's data collection, but I think we can assume some goodwill here for now.

Any uploaders not interested in this function can simply use "guest" to have the same experience as they do now. 

Pros
- no storage of sensitive information (email / password)
- low implementation cost
- gives us the ability to separate analytics triggers from uploads (allows other features to build out)

Cons
- high tamper-ability

Happy to hear everyone's thoughts!

**Comment by jclauneuro** [comment]
I like the idea of unique identifiers! Is there any way to link a login with some other type of account that many researchers/scientists have? Say how the Open Science Framework (osf.io) can use ORCID ids? But also allowing for guest uploaders so that experience is not limited.

**Comment by kaitj** [comment]
I like the idea of implementing ORCID ids as a login. One of the security things brought up was the responsibility of storing user passwords if we were to have a login. Wonder if ORCID ids gets around that.

**Comment by ostanley** [comment]
I'll look into orcid/osf logins and report back. It'd definitely be easier on our end to outsource that.

**Comment by ostanley** [comment]
https://members.orcid.org/api/tutorial/get-orcid-id

TLDR: the public API is free and has the functionality to allow for orcid login and authorized orcid id This seems like a strong option as the login and authentication is done on their side. What do people think?

**Comment by kaitj** [comment]
I like this. We can store the recommended data (last section).

**Comment by Park-Patrick** [comment]
seems like a great option!

**Comment by ostanley** [comment]
I'll play around with this then, get a working login and return on a separate page from the site. Then we can place it whenever we decide where/when to implement id saving.

**Comment by kaitj** [comment]
Going to be switching to orcid login

**Comment by kaitj** [comment]
Additional potential login is to make use of github logins. Need to look in whether passwords are stored (which would rule this out), or if everything is handled on Git's side.

**Comment by ostanley** [comment]
github logins do not expose pwds (not sure if this was looked into or not). Need to discuss which is preferred as solutions appear similar. email is kinda default with orcid and username is default with github so which is appropriate.

**Comment by kaitj** [comment]
Just leaving my thoughts here so I don't forget. Ultimately, I think what is more likely to be used? I personally prefer git logins, but if orcid is more common, then orcid may be the better bet. Or do we include both and treat them as separate users if the username doesn't match the email?

**Comment by jclauneuro** [comment]
+1 re github account integration. I think it would be reasonable to expect that afids users would either already have a github account or would be willing to create one to keep track of the annotations that they are uploading. For others, they would still be able to validate anonymously.

**Comment by kaitj** [comment]
Revisiting this to drop a potentially useful links for implementation of Github OAuth with Flask:

https://github.com/singingwolfboy/flask-dance

One thing that stands out from a quick glance, is whether we want to store the access tokens? From the README:

```
By default, OAuth access tokens are stored in Flask's session object. This means that if the user ever 
clears their browser cookies, they will have to go through the OAuth dance again
```

If we do want to store the token, they recommend a database or another persistent storage.

**Comment by kaitj** [comment]
With the introduction of databases, the tables have a column for storing the unique identifiers so that we can match uploads (if save option is selected) to uploader. Something to circle back to as the next feature.

**Comment by kaitj** [comment]
Unique identifiers has now been implemented via orcid oauth
